pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8926755?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8926755?buildTab=artifacts#/) on master @ [cf14ad694ee562676f53e36fa8495206c3aed61f](https://github.com/cockroachdb/cockroach/commits/cf14ad694ee562676f53e36fa8495206c3aed61f):


```
        delete {source=1,target=2}
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3334/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts:127:
        state [1 args]
        <no input to command>
        ----
        /{Min-System/NodeLiveness}                 ttl_seconds=3600 num_replicas=5
        /System/NodeLiveness{-Max}                 ttl_seconds=600 num_replicas=5
        ...
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3334/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts:134:
        release [1 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3334/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts:137:
        release [1 args]
        <no input to command>
        ----
    datadriven_test.go:124: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3334/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts:140:
         
        expected:
        delete /Table/10{6-7}
        upsert /Table/10{6-7}                      range default
        delete /Table/10{7-8}
        upsert /Table/10{7-8}                      range default
        
        found:
        delete /Table/10{6-7}
        upsert /Table/10{6-7}                      protection_policies=[{ts: 4}]
        delete /Table/10{7-8}
        upsert /Table/10{7-8}                      protection_policies=[{ts: 4}]
        delete /Table/10{6-7}
        upsert /Table/10{6-7}                      range default
        delete /Table/10{7-8}
        upsert /Table/10{7-8}                      range default
I230305 07:17:29.933256 6856383 testutils/testcluster/testcluster.go:148  [T1] 2905  TestCluster quiescing nodes
W230305 07:17:29.933747 6887737 jobs/registry.go:1080  [T1,n1] 2906  canceling all adopted jobs due to stopper quiescing
I230305 07:17:29.934606 6872550 server/start_listen.go:103  [T1,n1] 2907  server shutting down: instructing cmux to stop accepting
I230305 07:17:29.935300 7102142 jobs/registry.go:1495  [T1,n1] 2908  KEY VISUALIZER job 100: stepping through state succeeded
I230305 07:17:29.936114 6887421 sql/stats/automatic_stats.go:507  [T1,n1] 2909  quiescing auto stats refresher
W230305 07:17:29.937715 7510389 kv/kvserver/intentresolver/intent_resolver.go:826  [T1] 2910  failed to gc transaction record: could not GC completed transaction anchored at /Table/31/1/1/0: node unavailable; try another peer
W230305 07:17:29.941171 6881647 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 2911  exiting heartbeat loop
W230305 07:17:29.941313 6881647 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 2912  exiting heartbeat loop
E230305 07:17:29.941436 6881647 server/server_sql.go:477  [T1,n1] 2913  failed to run update of instance with new session ID: node unavailable; try another peer
E230305 07:17:29.948053 6887814 jobs/registry.go:1064  [T1,n1] 2914  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: failed to read query result: query execution canceled
W230305 07:17:29.951681 7102142 jobs/adopt.go:457  [T1,n1] 2915  could not clear job claim: clear-job-claim: node unavailable; try another peer
W230305 07:17:29.953911 7129191 jobs/adopt.go:457  [T1,n1] 2916  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/101,/Table/15/1/102), [txn: 41f3b112], [can-forward-ts] RPC error: node unavailable; try another peer
    --- FAIL: TestDataDriven/protectedts (7.79s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25036

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts
@@ -130,6 +130,13 @@ state limit=2
 /System/NodeLiveness{-Max}                 ttl_seconds=600 num_replicas=5
 ...
 
+# Take another look at the remaining protected spans.
+state offset=57
+----
+...
+/Table/10{6-7}                             protection_policies=[{ts: 3} {ts: 4}]
+/Table/10{7-8}                             protection_policies=[{ts: 3} {ts: 4}]
+
 # Release the schema object records.
 release record-id=3
 ----
@@ -137,9 +144,9 @@ release record-id=3
 release record-id=4
 ----
 
-mutations
+# Observe that they're no longer protected.
+state offset=57
 ----
-delete /Table/10{6-7}
-upsert /Table/10{6-7}                      range default
-delete /Table/10{7-8}
-upsert /Table/10{7-8}                      range default
+...
+/Table/10{6-7}                             range default
+/Table/10{7-8}                             range default

